### ISSUE TITLE
CA2025: it doesn't trigger if unawaited task argument is IDisposable

### ISSUE THREAD

**Issue by jozkee** [dotnet/roslyn-analyzers#7775]
```cs
public static Task DoSomethingAsync()
{
    // Using statements and using blocks can both be violations.
    using (var disposable = DisposableThing())
    {
        return DoSomethingInternalAsync(disposable);
    }
}

public async Task DoThingsAsync()
{
    var disposable = DisposableThing();
    var task = DoSomethingInternalAsync(disposable);
    // More code here.
    disposable.Dispose();
    // It's a violation if arguments are disposed before the task is awaited.
    await task.ConfigureAwait(false);
}

static MemoryStream DisposableThing() => null!;

static Task DoSomethingInternalAsync(IDisposable disposable) { return null; } // Does not trigger
//static Task DoSomethingInternalAsync(Stream disposable) { return null; } // Trigger
```

CA2025 is disabled by default so you need to add an .editorcofig to repro this:
```
root = true

[*.cs]
# CA2025: Do not use 'Task' with 'using'
dotnet_diagnostic.CA2025.severity = error
```

cc @steveberdy
